### PR TITLE
DAOS-14245 gurt: Move print_backtrace to gurt.

### PR DESCRIPTION
**Pull request by ashleypittman**
Make this function available to any daos code, and enabled by
default in the engine and dfuse.

Required-githooks: true
Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/gurt/signals.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+void
+d_signal_register()
+{
+	bool enabled = register_handler;
+
+	if (registered)
+		return;
+
+	d_getenv_bool("DAOS_SIGNAL_REGISTER", &enabled);
```
Comment (ashleypittman) [inline src/gurt/signals.c:135]: @frostedcmos is there a list I need to add this to?

File: src/gurt/signals.c [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		/* XXX will old handler get accurate siginfo_t/ucontext_t ?
+		 * we may prefer to call it with the same params we got ?
+		 */
+		raise(signo);
+	}
+
+	memset(&old_handlers[signo], 0, sizeof(struct sigaction));
+}
+
+bool register_handler = false;
```
Comment (jolivier23) [inline src/gurt/signals.c:118]: Should these be static?

**Review by jolivier23 — CHANGES_REQUESTED** [review]
sorry, changing approval.  I think those variables can and should be static.

File: src/gurt/signals.c (hunk 2 above)
Comment (ashleypittman) [reply src/gurt/signals.c:118]: Yes.  Will wait for Alexs response and repush.

Comment (ashleypittman) [reply src/gurt/signals.c:118]: I'm slightly annoyed these don't cause a build failure, do we not check for exported symbols from the library any more?  Perhaps modern compiler flags is hiding them though?   Anyway, will update.

File: src/gurt/signals.c (hunk 1 above)
Comment (frostedcmos) [reply src/gurt/signals.c:135]: > @frostedcmos is there a list I need to add this to?

We have a dump list used in dump_envariables() at init time for debug purposes, but we typically only add cart or mercury/ofi level envariables there.

File: src/include/gurt/common.h [hunk 3]
```diff
@@ -987,6 +987,19 @@ int d_vec_pointers_init(struct d_vec_pointers *pointers, uint32_t cap);
 void d_vec_pointers_fini(struct d_vec_pointers *pointers);
 int d_vec_pointers_append(struct d_vec_pointers *pointers, void *pointer);
 
+/** Change the default setting for if a signal handler should be installed in cart_init()
```
Comment (frostedcmos) [inline src/include/gurt/common.h:990]: minor: crt_init() not cart_init()

File: src/gurt/signals.c (hunk 1 above)
Comment (ashleypittman) [reply src/gurt/signals.c:135]: Found, and done.

**Review by frostedcmos — DISMISSED** [review]

**Review by frostedcmos — APPROVED** [review]

File: src/gurt/signals.c (hunk 2 above)
Comment (jolivier23) [reply src/gurt/signals.c]: right now, most of our libraries export all symbols that are not static.  Only libioil, I think changes this behavior by setting that PUBLIC api flag.  We should probably address that.

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**